web/pages/tournaments/index.tsx

```diff
@@ -107,130 +107,139 @@ const tourneys: Tourney[] = [
+          {title}
+        </h2>
+        <Row className="my-2 items-center gap-4 whitespace-nowrap rounded-full bg-gray-200 px-6">
+          {!!award && <span className="flex items-center">🏆 {award}</span>}
+          {!!ppl && (
+            <span className="flex items-center gap-1">
+              <UsersIcon className="h-4" />
+              {ppl}
+            </span>
+          )}
+          {endTime && (
+            <DateTimeTooltip time={endTime} text="Ends">
               <span className="flex items-center gap-1">
-                <UsersIcon className="h-4" />
-                {ppl}
+                <ClockIcon className="h-4" />
+                {dayjs(endTime).format('MMM D')}
               </span>
-            )}
-            {endTime && (
-              <DateTimeTooltip time={endTime.valueOf()} text="Ends">
-                <span className="flex items-center gap-1">
-                  <ClockIcon className="h-4" />
-                  {endTime.format('MMM D')}
-                </span>
-              </DateTimeTooltip>
-            )}
-          </Row>
+            </DateTimeTooltip>
+          )}
+        </Row>
+      </a>
+    </Link>
+  )
+}
+
+const ImageCarousel = (props: { images: MarketImage[]; url: string }) => {
+  const { images, url } = props
+  return (
+    <Carousel className="-mx-4 mt-4 sm:-mx-10">
+      <div className="shrink-0 sm:w-6" />
+      {images.map(({ marketUrl, image }) => (
+        <a key={marketUrl} href={marketUrl} className="hover:brightness-95">
+          <NaturalImage src={image} />
         </a>
-      </Link>
-      <span>{blurb}</span>
-      <Carousel className="-mx-4 mt-2 sm:-mx-10">
-        <div className="shrink-0 sm:w-6" />
-        {markets.length ? (
-          markets.map((m) => (
-            <ContractCard
-              contract={m}
-              hideGroupLink
-              className="mb-2 max-h-[200px] w-96 shrink-0"
-              questionClass="line-clamp-3"
-              trackingPostfix=" tournament"
-            />
-          ))
-        ) : (
-          <>
-            {images?.map(({ marketUrl, image }) => (
-              <a href={marketUrl} className="hover:brightness-95">
-                <NaturalImage src={image} />
-              </a>
-            ))}
-            <SiteLink
-              className="ml-6 mr-10 flex shrink-0 items-center text-indigo-700"
-              href={url}
-            >
-              See more
-            </SiteLink>
-          </>
-        )}
-      </Carousel>
-    </div>
+      ))}
+      <SiteLink
+        className="ml-6 mr-10 flex shrink-0 items-center text-indigo-700"
+        href={url}
+      >
+        See more
+      </SiteLink>
+    </Carousel>
+  )
+}
+
+const MarketCarousel = (props: { slug: string }) => {
+  const { slug } = props
+  const q = contractsByGroupSlugQuery(slug)
+  const { allItems, getNext, isLoading } = usePagination({ q, pageSize: 12 })
```

Comment:
I doubled pagesize to 12 so that on first click, we don't require a load.

(The load looks really janky atm because the entire contract cards disappear on the next load; would like to get this fixed this later, but not high priority)

Reply:
Also really janky because the load resets your scroll position in the carousel....

Might not even be worth paginating, but rather have a final card that's "See all" that you click on and takes you to the entire page.

Reply:
It sounds like there's something going wrong here. It worked fine when I was using it.